packages/@ember/-internals/glimmer/lib/helper.ts

```diff
@@ -145,9 +151,20 @@ class ClassicHelperManager implements HelperManager<ClassicHelperStateBucket> {
     hasDestroyable: true,
   });
 
+  constructor(public owner: Owner | undefined) {
+    CLASSIC_HELPER_MANAGERS.add(this);
+  }
+
   createHelper(definition: ClassHelperFactory, args: Arguments) {
+    let ownerInjection;
+
+    if (definition.class === undefined) {
+      ownerInjection = {};
+      setOwner(ownerInjection, this.owner!);
+    }
+
     return {
-      instance: definition.create(),
+      instance: definition.create(ownerInjection),
       args,
     };
```

Comment:
we can't call `new definition` because it extends from `EmberObject`, that would throw an error. We could create a single `ownerInjection` object for the entire manager though, so there are fewer of them

Reply:
@pzuraq - Can you link me to where we error for that? A quick test locally suggests that there is no error or deprecation when calling `new Ember.Object()`...

Reply:
I assume you mean this one? 

https://github.com/emberjs/ember.js/blob/v3.16.10/packages/@ember/-internals/runtime/lib/system/core_object.js#L288-L299

Which you removed in https://github.com/emberjs/ember.js/pull/19088...